### PR TITLE
Improve govendor testing

### PR DESCRIPTION
**Pull request by sapk**
Use `govendor list +outside +unused` for finding missing or unused deps and `govendor status` for catching modified vendor.

Ref : #1620

### CODE REVIEW

**Comment by sapk** [comment]
This should failed at CI and be ok after go-gitea/gitea#1620. So we could merge it after.

**Comment by appleboy** [comment]
@sapk rebase master. I already merged #1620

**Comment by lunny** [comment]
@sapk but seems CI is not failed.

**Comment by sapk** [comment]
@lunny I need to count line of `govendor list +outside +unused` since it return 0 every time.

File: vendor/vendor.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"comment": "",
-	"ignore": "test",
+	"ignore": "test appengine",
```
Comment (bkcsoft) [inline vendor/vendor.json:3]: Why was this added?

Comment (sapk) [reply vendor/vendor.json:3]: Because we have un-vendored deps relative to appengine. I think that it's a sub deps but I haven't found/search for it. I don't even know if we (plan to) support appengine. So this could be a good place to discuss about it ^^

File: Makefile [hunk 2]
```diff
@@ -84,7 +84,13 @@ test-vendor:
 	@hash govendor > /dev/null 2>&1; if [ $$? -ne 0 ]; then \
 		go get -u github.com/kardianos/govendor; \
 	fi
-	govendor status +outside +unused  || exit 1
+	govendor list +unused | tee /tmp/wc-gitea-unused
```
Comment (bkcsoft) [inline Makefile]: Please use `$TMPDIR` instead (same for the next command-block)

**Review by bkcsoft — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (sapk) [reply Makefile]: The var TMPDIR (or TEMPDIR) seems  to not be set when I run it via `drone exec`. I will add `TMPDIR := $(shell mktemp -d)`

**Comment by lunny** [comment]
LGTM

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

**Comment by appleboy** [comment]
@bkcsoft waiting for your confirm.

**Review by bkcsoft — APPROVED** [review]
